src/azul/uuids.py

```diff
@@ -0,0 +1,77 @@
+import uuid
+
+
+class InvalidUUIDError(Exception):
+
+    def __init__(self, entity_id: str):
+        super().__init__(f'{entity_id} is not a valid uuid.')
+
+
+class InvalidUUIDPrefixError(Exception):
+
+    def __init__(self, prefix: str):
+        super().__init__(f'{prefix} is not a valid uuid prefix.')
+
+
+def validate_uuid(uuid_str: str, allow_uppercase: bool = False) -> uuid.UUID:
+    """
+    >>> validate_uuid('foo')
+    Traceback (most recent call last):
+    ...
+    azul.uuids.InvalidUUIDError: foo is not a valid uuid.
+
+    >>> validate_uuid('aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa')
```

Comment:
This is a version 10 UUID which hasn't been standardized. We should be stricter and only allow UUID versions that have been formally specified.